providers/filesystem/filesystem.go

```diff
@@ -53,6 +53,12 @@ func NewBucket(rootDir string) (*Bucket, error) {
 // Iter calls f for each entry in the given directory. The argument to f is the full
 // object name including the prefix of the inspected directory.
 func (b *Bucket) Iter(ctx context.Context, dir string, f func(string) error, options ...objstore.IterOption) error {
+	select {
```

Comment:
[nit] I think you can simplify this into:

```
if ctx.Err() != nil {
  return ctx.Err()
}
```

Reply:
Good point, resolved in 666b242.